### ISSUE TITLE
search.php only partly translatable

### ISSUE THREAD

**Issue by Langveisfra** [meeting-room-booking-system/mrbs-code#3502]
The top and the bottom of the table showing the result of a search is displayed in English, even though the MRBS is forced to show another language. The terms that are not translated are present in the lang.xx-file.

The error was not present in MRBS version 1.4.9, but occurs in 1.11.3.

Running:
10.3.37-MariaDB
Apache/2.4.57 (Unix)
PHP | 8.2.5

![image](https://github.com/meeting-room-booking-system/mrbs-code/assets/132160113/8f4c32cb-81f8-4735-bf98-84cb092e2714)

**Comment by Langveisfra** [comment]
Showing the non-translated items:
![image](https://github.com/meeting-room-booking-system/mrbs-code/assets/132160113/46877e08-8953-45c8-b7d7-a93cd04f908b)

**Comment by campbell-m** [comment]
I think it actually works OK in 1.11.3, but not in the latest development code.

The problem is that in the latest development code, which uses the latest DataTables library, there is only a sv-SE.json file.  (In 1.11.3 and earlier, where the previous versions of DataTables used a different set of language files, there was a Swedish.lang file).  In the short term you could fix your system by copying jquery/datatables/language/sv-SE.json to sv.json.

As for the proper fix I think there are a number of options:

1. Submit a PR to the DataTables team to provide a sv.json file; and/or
2. Allow `$override_locale` to be an array; and/or
3. Change the algorithm that searches for the best language file to fall back to 'sv-SE' if it cannot find 'sv'.

The last two options require a bit more thought though.

**Comment by campbell-m** [comment]
I've raised [an issue](https://github.com/DataTables/Plugins/issues/576) on the DataTables Plugins.

**Comment by campbell-m** [comment]
The other workaround, assuming that at the moment you have `$override_locale = 'sv';` is to set `$override_locale = 'sv-SE';` instead.

**Comment by Langveisfra** [comment]
I put  $override_locale = 'sv-SE'; in the config.inc.php file without success. Should it be placed elsewhere?

**Comment by campbell-m** [comment]
No, that's where it should go.  I'm not sure whether you are running 1.11.3 or the latest development code.  If you are running 1.11.3 I would recommend upgrading to the latest development code.  You can download the files by following the green Code button on [this page](https://github.com/meeting-room-booking-system/mrbs-code).  Treat it like an upgrade.

**Comment by Langveisfra** [comment]
The DataTables version included in MRBS version 1.11.3 is 1.10.25, and in the jquery/datatables/language - directory there actually is a Swedish.lang - file.
Did I misunderstand something?

**Comment by campbell-m** [comment]
No, I misunderstood. It means you are in fact running 1.11.3. I recommend that you upgrade to the latest development code as described above.

**Comment by Langveisfra** [comment]
The not-translates terms resides in the files of this folder: mrbs/jquery/datatables/language/[Language].lang
The language files seems not to be accessed, with or without stating $override_locale = '[languagecode]'; in the config.inc.php file

**Comment by campbell-m** [comment]
Have you upgraded to the latest development code?

**Comment by Langveisfra** [comment]
No, not yet. Maybe I'll try it tonight. Does it alter the sql-database so that I can not revert to the ordinary 1.11.3 release?
Have you tried the dev. release with success regarding table text translation?

**Comment by campbell-m** [comment]
No, it won't alter the database.  Yes, I've tested it successfully with `$override_locale = 'sv-SE';`.

**Comment by Langveisfra** [comment]
Strange things. I did install the dev. version, leaving the $override_locale = 'sv-SE'; as the only language pref. (Commented  out $faqfilelang - $disable_automatic_language_changing - $default_language_tokens) 
All is still showing in Swedish - except from the tables top and bottom text, as circled in red above.

**Comment by Langveisfra** [comment]
Strange things. I did install the dev. version, leaving the $override_locale = 'sv-SE'; as the only language pref. (Commented  out $faqfilelang - $disable_automatic_language_changing - $default_language_tokens) 
All is still showing in Swedish - except from the tables top and bottom text, as circled in red above.

**Comment by Langveisfra** [comment]
After restaring the webserver (apache running on Synology) it worked. Thank you!

**Comment by Langveisfra** [comment]
But alas, after renaming the web folder containing the MRBS's web files - it's back to normal: All Swedish except the DataTables text, which still is English.

**Comment by campbell-m** [comment]
Try doing a hard refresh of your browser, or else using a different browser.  It's possible that your browser is using a cached version of the JavaScript file.

**Comment by Langveisfra** [comment]
New browser (that never have visited the site on beforehand) = no luck. Downgrading to MRBS 1.11.2 = huge success :-) It seems to me that this bug (failing to consistently localize text elements in DataTables) was introduced in version 1.11.3 and still persists in det dev version.
[The system demo](https://mrbs.sourceforge.io/demo.php), running version 1.11.1 also shows translated text fine.

**Comment by campbell-m** [comment]
What are your settings of `$disable_automatic_language_changing`, `$default_language_tokens` and `$override_locale`?

Also, what are your browser's language preferences (as shown on the MRBS Help page)?

**Comment by campbell-m** [comment]
Can you try the attached version of js/datables.js.php with the latest development version?  You'll need to unzip the file first.
[datatables.js.zip](https://github.com/meeting-room-booking-system/mrbs-code/files/12579232/datatables.js.zip)

**Comment by jberanek** [comment]
> New browser (that never have visited the site on beforehand) = no luck. Downgrading to MRBS 1.11.2 = huge success :-) It seems to me that this bug (failing to consistently show localized text elements in DataTables) was introduced in version 1.11.3 and still persists in det dev version. [The system demo](https://mrbs.sourceforge.io/demo.php), running version 1.11.1 also shows translated text fine.

Hi, I've solved the "issue" of the demo site having an old version of MRBS, it is not running 1.11.3.

This site is running the tip of the 'main' branch up until just now, when I pulled the latest code:
https://mrbsdemo.uk.to/mrbs-tip/

**Comment by campbell-m** [comment]
I note that the demo site running 1.11.3 works properly when I have my browser language set to 'sv', but the tip doesn't (as I would expect).

**Comment by Langveisfra** [comment]
> Can you try the attached version of js/datables.js.php with the latest development version? You'll need to unzip the file first. [datatables.js.zip](https://github.com/meeting-room-booking-system/mrbs-code/files/12579232/datatables.js.zip)

Tried your attached file. Still English titles above and below the table.

**Comment by Langveisfra** [comment]
> I note that the demo site running 1.11.3 works properly when I have my browser language set to 'sv', but the tip doesn't (as I would expect).

Same result for me. One works, the other doesn't.

**Comment by campbell-m** [comment]
I've now added in [56b28ca](https://github.com/meeting-room-booking-system/mrbs-code/commit/56b28caeb0650a03c8101e6f016c26c6e0e66e95) some basic language versions which should make the tip, when updated, work with 'sv'.

**Comment by campbell-m** [comment]
> Tried your attached file. Still English titles above and below the table.

Did you do a hard refresh of the browser?

Can you have a look in your browser's development tools and see if there are any errors in (a) the console and (b) the network tabs?

Also, if your site is on the internet and you are willing to give me a link, could you send me a PM?  If not, I quite understand.

**Comment by campbell-m** [comment]
> What are your settings of $disable_automatic_language_changing, $default_language_tokens and $override_locale?

> Also, what are your browser's language preferences (as shown on the MRBS Help page)?

Don't know if you had a chance to look at these?

**Comment by Langveisfra** [comment]
How do I send a PM in GitHub?

**Comment by campbell-m** [comment]
Good point, I don't think you can.  But you could email me: cimorrison at users dot sourceforge dot net

**Comment by Langveisfra** [comment]
Campell-m: I have sent you a message.

**Comment by Langveisfra** [comment]
Campbell-m: I have sent you a message.

**Comment by campbell-m** [comment]
OK, it hasn't arrived yet, but I'll look out for it.

**Comment by Langveisfra** [comment]
I posted it within the message system of sourceforge. Sourceforge does not seem to include a regular e-mail system.

**Comment by campbell-m** [comment]
Now working.  It seems that the problem was caused by not having the most up-to-date language.inc file on the web server.
